### ISSUE TITLE
Multilevel object with nested object/collection ignoring it's parent couldn't be persisted

### ISSUE THREAD

**Issue by Deilan** [mbdavid/LiteDB#1055]
TL;DR: [Repository with the tests](https://github.com/deilan-issues-samples/LiteDB.Issues). More specifically:
- [NestedObjectMappingTests.cs](https://github.com/deilan-issues-samples/LiteDB.Issues/blob/master/LiteDB.Issues.Tests/Mapping/NestedObjectMappingTests.cs)
- [NestedCollectionMappingTests.cs](https://github.com/deilan-issues-samples/LiteDB.Issues/blob/master/LiteDB.Issues.Tests/Mapping/NestedCollectionMappingTests.cs)

Models:
```csharp
public class Customer
{
    public int Id { get; set; }
    public Order Order { get; set; }
}
public class Order
{
    public Customer Customer { get; set; }
}
```

Arrange:
```csharp
var customer = new Customer();
var order = new Order();
customer.Order = order;
order.Customer = customer;
```

Act:
```csharp
BsonMapper.Global.Entity<Customer>().Ignore(x => x.Order.Customer);

var customersCollection = liteDatabase.GetCollection<Customer>();

customersCollection.Insert(customer);
var persistedCustomer = customersCollection.FindById(customer.Id);
```

Assert:
```csharp
Assert.Null(persistedCustomer.Order.Customer);
```

Actually `BsonMapper.Global.Entity<Customer>().Ignore(x => x.Order.Customer);` throws
`ArgumentNullException`.

The reason I can't configure `Order` entity itself to ignore `Customer` is because I use it as a nested property in multiple parent entities. The entities' names were chosen simply for demo purposes.

**Comment by lbnascimento** [comment]
Hi! With the objective of organizing our issues, we are closing old unsolved issues. Please check the latest version of LiteDB and open a new issue if your problem/question/suggestion still applies. Thanks!
